src/main/java/edu/ie3/datamodel/io/source/sql/SqlWeatherSource.java

```diff
@@ -146,6 +159,40 @@ public SqlWeatherSource(
     return Optional.of(timeBasedValues.get(0));
   }
 
+  /**
+   * Determine the corresponding the database column name based on the provided factory field
+   * parameter name. Needed to support camel as well as snake case database column names.
+   *
+   * @param factoryColumnName the name of the field parameter set in the entity factory
+   * @param connector the sql connector of this source
+   * @param weatherTableName the table name where the weather is stored
+   * @return the column name that corresponds to the provided field parameter or an empty optional
+   *     if no matching column can be found
+   */
+  private Optional<String> dbColumnName(
```

Comment:
I would prefer a method name with a verb in it. E.g.
```suggestion
  private Optional<String> determineDbColumnName(
```